au3/libraries/au3-math/CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+# SPDX-License-Identifier: GPL-3.0-only
+# MuseScore-CLA-applies
+#
+# MuseScore
+# Music Composition & Notation
+#
+# Copyright (C) 2021 MuseScore BVBA and others
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# AU3 Math Library
+# A library of mathematical utilities and manipulation of samples
+
+set(SOURCES
+    Biquad.cpp
+    EBUR128.cpp
+    SampleFormat.cpp
+    SampleCount.cpp
+    Resample.cpp
+    Dither.cpp
+    InterpolateAudio.cpp
+    Matrix.cpp
+    PitchName.cpp
+)
+
+set(HEADERS
+    Biquad.h
+    EBUR128.h
+    SampleFormat.h
+    SampleCount.h
+    Resample.h
+    Dither.h
+    InterpolateAudio.h
+    Matrix.h
+    LinearFit.h
+    PitchName.h
+    Reverb_libSoX.h
+    RoundUpUnsafe.h
+    float_cast.h
+    Gain.h
+)
+
+add_library(au3-math STATIC ${SOURCES} ${HEADERS})
+
+target_include_directories(au3-math
+    PUBLIC
+        # Allow includes like: #include "au3-math/SampleFormat.h"
+        ${CMAKE_CURRENT_SOURCE_DIR}/..
+        # For internal includes within au3-math (e.g., "SampleFormat.h")
+        ${CMAKE_CURRENT_SOURCE_DIR}
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/../lib-preferences
+        ${CMAKE_CURRENT_SOURCE_DIR}/../lib-strings
+        ${CMAKE_CURRENT_SOURCE_DIR}/../lib-utility
+        ${CMAKE_CURRENT_SOURCE_DIR}/../lib-components
+        # For soxr.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/../../lib-src/libsoxr/src
+)
+
+target_compile_definitions(au3-math
+    PUBLIC
+        -DMATH_API=
+        # wxWidgets base restrictions (no GUI)
+        -DwxUSE_GUI=0
+        -D_WX_APP_H_BASE_
```

Comment:
do we need all of that in all au3 libraries? Maybe we can use a macro to register all libraries in the same way